src/createServer.js

```diff
@@ -1,11 +1,54 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
+const { UserController } = require('./controllers/users.js');
+const { ExpensesController } = require('./controllers/expenses.js');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+
+  app.use(express.json());
+
+  const userController = new UserController();
+  const expensesController = new ExpensesController();
+
+  app.post('/users', userController.add.bind(userController));
+
+  app.get('/users', userController.getAll.bind(userController));
+
+  app.get('/users/:id', userController.getOne.bind(userController));
+
+  app.patch('/users/:id', userController.change.bind(userController));
+
+  app.delete('/users/:id', userController.remove.bind(userController));
+
+  app.post('/expenses', (req, res) => {
+    const { userId } = req.body;
+    const foundUser = userController.usersService.getUserById(userId);
+
+    if (!foundUser) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const newExpense = expensesController.expensesService.create(req.body);
+
+    res.status(201);
+    res.send(newExpense);
```

Comment:
You've created a bunch of problems when decided to use classes for your app. As for me the better approach is to have all methods as separate functions: no `this` problem, better reusability and flexibility. For example, this function could be moved to Expenses controller too.